validator/src/admin_rpc_service.rs

```diff
@@ -263,6 +288,37 @@ impl AdminRpc for AdminRpcImpl {
         Ok(())
     }
 
+    fn set_block_engine_config(
+        &self,
+        meta: Self::Metadata,
+        auth_service_addr: String,
+        backend_addr: String,
+        trust_packets: bool,
+    ) -> Result<()> {
+        debug!("set_block_engine_config request received");
+
+        if !(backend_addr.contains("http") || backend_addr.is_empty()) {
```

Comment:
what if client supplies a string that passes this check but contains invalid URI chars?

Reply:
updated validation functions and used here as suggested